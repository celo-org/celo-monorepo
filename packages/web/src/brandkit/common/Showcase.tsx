import * as React from 'react'
import FadeIn from 'react-lazyload-fadein'
import { Image, ImageURISource, StyleSheet, Text, View } from 'react-native'
import { brandStyles } from 'src/brandkit/common/constants'
import DownloadButton from 'src/brandkit/common/DownloadButton'
import AspectRatio from 'src/shared/AspectRatio'
import Spinner from 'src/shared/Spinner'
import { colors, fonts, standardStyles } from 'src/styles'

interface Props {
  name: string
  description: string
  preview?: ImageURISource
  uri: string
  ratio: number
  loading: boolean
  size: number | '100%'
}

export default React.memo(function Showcase({
  name,
  description,
  preview,
  loading,
  uri,
  size,
  ratio,
}: Props) {
  return (
    <View
      style={[
        brandStyles.gap,
        standardStyles.elementalMarginTop,
        styles.container,
        { width: size },
      ]}
    >
      <FadeIn>
        {(load) => (
          <View style={styles.previewContainer}>
            <AspectRatio ratio={ratio}>
              {loading ? (
                <Spinner color={colors.primary} size="small" />
              ) : (
                <Image
                  onLoadEnd={load}
                  resizeMode="contain"
                  accessibilityLabel={`Preview of ${name}`}
                  source={preview}
                  style={standardStyles.image}
                />
              )}
            </AspectRatio>
          </View>
        )}
      </FadeIn>
      <View style={styles.text}>
        <Text style={[fonts.h6, styles.title]}>{name}</Text>
        <Text style={fonts.legal}>{description}</Text>
      </View>
      <DownloadButton uri={uri} />
    </View>
  )
})

const styles = StyleSheet.create({
  title: { marginVertical: 5 },
  previewContainer: {
    marginVertical: 20,
    marginRight: 40,
  },
  text: { flex: 1 },
  pullStart: { paddingLeft: 0 },
  container: {
    justifyContent: 'space-between',
  },
})
