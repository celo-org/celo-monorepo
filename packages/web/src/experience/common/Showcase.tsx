import * as React from 'react'
import { Image, StyleSheet, Text, View } from 'react-native'
import Fade from 'react-reveal/Fade'
import DownloadButton from 'src/experience/brandkit/DownloadButton'
import { AssetTypes } from 'src/experience/brandkit/tracking'
import { brandStyles } from 'src/experience/common/constants'
import AspectRatio from 'src/shared/AspectRatio'
import Spinner from 'src/shared/Spinner'
import { colors, fonts, standardStyles } from 'src/styles'

interface Props {
  name: string
  description: string
  preview?: string
  uri: string
  ratio: number
  loading: boolean
  size: number | '100%'
  assetType: AssetTypes
  variant?: 'circle' | 'circle-white' | 'circle-black'
}

export default React.memo(function Showcase({
  name,
  description,
  preview,
  loading,
  uri,
  assetType,
  size,
  ratio,
  variant,
}: Props) {
  const trackingData = React.useMemo(() => ({ name: `${name} ${assetType}`, type: assetType }), [
    name,
    assetType,
  ])
  const titleStyle = [fonts.h6, styles.title]
  return (
    <View style={[brandStyles.gap, standardStyles.elementalMarginTop, { width: size }]}>
      <Fade duration={FADE_MS}>
        <View>
          <View style={styles.previewContainer}>
            <AspectRatio ratio={ratio}>
              {loading ? (
                <Spinner color={colors.primary} size="small" />
              ) : (
                <Image
                  resizeMode="contain"
                  accessibilityLabel={`Preview of ${name}`}
                  source={{ uri: preview }}
                  style={[standardStyles.image, styles[`variant-${variant}`]]}
                />
              )}
            </AspectRatio>
          </View>
          <View style={styles.text}>
            {name && <Text style={titleStyle}>{name.trimLeft()}</Text>}
            {description && <Text style={fonts.legal}>{description}</Text>}
          </View>
          {uri && <DownloadButton uri={uri} trackingData={trackingData} />}
        </View>
      </Fade>
    </View>
  )
})

const FADE_MS = 400

const styles = StyleSheet.create({
  title: { marginVertical: 5 },
  previewContainer: {
    marginVertical: 20,
    marginRight: 40,
  },
  text: { flex: 1 },
  pullStart: { paddingLeft: 0 },
  'variant-circle': {
    borderRadius: 60,
  },
  'variant-circle-white': {
    borderRadius: 60,
    boxShadow: `inset 0 0 0 1px ${colors.placeholderGray}`,
  },
  'variant-circle-black': {
    borderRadius: 60,
    backgroundColor: '#000000',
  },
})
