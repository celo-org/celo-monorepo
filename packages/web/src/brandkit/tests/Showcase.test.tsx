import * as React from 'react'
import * as renderer from 'react-test-renderer'
import Showcase from 'src/brandkit/common/Showcase'

describe('Showcase', () => {
  it('renders', () => {
    const showCase = renderer
      .create(
        <Showcase
          name={'test'}
          description="Trust, Attest, Verify"
          loading={false}
          size={100}
          uri="example.com"
        />
      )
      .toJSON()
    expect(showCase).toMatchSnapshot()
  })
})
