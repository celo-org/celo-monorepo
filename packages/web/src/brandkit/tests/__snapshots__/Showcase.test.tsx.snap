// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Showcase renders 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxPack": "justify",
      "WebkitJustifyContent": "space-between",
      "justifyContent": "space-between",
      "marginLeft": "10px",
      "marginRight": "10px",
      "marginTop": "20px",
      "msFlexPack": "justify",
      "width": "100px",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "marginBottom": "20px",
        "marginRight": "40px",
        "marginTop": "20px",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "display": "block",
          "overflowX": "hidden",
          "overflowY": "hidden",
        }
      }
    >
      <div
        style={
          Object {
            "display": "block",
            "paddingBottom": "100%",
            "width": "100%",
          }
        }
      />
      <div
        style={
          Object {
            "bottom": "0px",
            "height": "100%",
            "left": "0px",
            "position": "absolute",
            "top": "0px",
            "width": "100%",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitFlexBasis": "auto",
              "flexBasis": "auto",
              "height": "100%",
              "msFlexPreferredSize": "auto",
              "overflowX": "hidden",
              "overflowY": "hidden",
              "width": "100%",
              "zIndex": 0,
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.00)",
                "backgroundPosition": "center",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "contain",
                "bottom": "0px",
                "height": "100%",
                "left": "0px",
                "position": "absolute",
                "right": "0px",
                "top": "0px",
                "width": "100%",
                "zIndex": -1,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitBoxFlex": 1,
        "WebkitFlexBasis": "0%",
        "WebkitFlexGrow": 1,
        "WebkitFlexShrink": 1,
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
        "msFlexNegative": 1,
        "msFlexPositive": 1,
        "msFlexPreferredSize": "0%",
      }
    }
  >
    <div
      className="css-text-76zvg2"
      dir="auto"
      style={
        Object {
          "color": "rgba(46,51,56,1.00)",
          "fontFamily": "Jost, futura-pt, futura, sans-serif",
          "fontSize": "16px",
          "fontWeight": "500",
          "lineHeight": "18px",
          "marginBottom": "5px",
          "marginTop": "5px",
          "textRendering": "geometricPrecision",
        }
      }
    >
      test
    </div>
    <div
      className="css-text-76zvg2"
      dir="auto"
      style={
        Object {
          "color": "rgba(46,51,56,1.00)",
          "fontFamily": "EB Garamond, eb-garamond, Garamond, serif",
          "fontSize": "16px",
          "lineHeight": "20px",
          "textRendering": "geometricPrecision",
        }
      }
    >
      Trust, Attest, Verify
    </div>
  </div>
  <div
    className="css-view-1dbjc4n"
    onClick={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onResponderGrant={[Function]}
    onResponderRelease={[Function]}
  >
    <a
      className="css-reset-4rbku5 css-cursor-18t94o4 css-text-76zvg2"
      data-focusable={true}
      dir="auto"
      href="example.com"
      role="link"
      style={
        Object {
          "WebkitAlignItems": "center",
          "WebkitBoxAlign": "center",
          "WebkitBoxFlex": 0,
          "WebkitBoxPack": "start",
          "WebkitFlexGrow": 0,
          "WebkitJustifyContent": "flex-start",
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "borderBottomColor": "rgba(0,0,0,0.00)",
          "borderLeftColor": "rgba(0,0,0,0.00)",
          "borderRightColor": "rgba(0,0,0,0.00)",
          "borderTopColor": "rgba(0,0,0,0.00)",
          "color": "rgba(53,208,127,1.00)",
          "cursor": "pointer",
          "display": "inline-flex",
          "flexGrow": 0,
          "fontFamily": "Jost, futura-pt, futura, sans-serif",
          "fontSize": "16px",
          "fontWeight": "500",
          "justifyContent": "flex-start",
          "lineHeight": "16px",
          "minWidth": "150px",
          "msFlexAlign": "center",
          "msFlexPack": "start",
          "msFlexPositive": 0,
          "opacity": 1,
          "outlineColor": "transparent",
          "paddingBottom": "10px",
          "paddingLeft": "0px",
          "paddingRight": "20px",
          "paddingTop": "10px",
          "textAlign": "center",
          "textRendering": "geometricPrecision",
          "width": "fit-content",
        }
      }
    >
      downloadAssetBtn
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "display": "inline-flex",
            "paddingLeft": "8px",
          }
        }
      >
        <svg
          fill="none"
          height={12}
          style={Object {}}
          viewBox="0 0 10 13"
          width={12}
        >
          <path
            d="M0 12H10"
            stroke="#35D07F"
            strokeWidth="1.5"
            style={Object {}}
          />
          <path
            d="M5 0V9"
            stroke="#35D07F"
            strokeWidth="1.5"
            style={Object {}}
          />
          <path
            d="M1 5L5 9L9 5"
            stroke="#35D07F"
            strokeWidth="1.5"
            style={Object {}}
          />
        </svg>
      </div>
    </a>
  </div>
</div>
`;
